scripts/partition_manager.py

```diff
@@ -244,61 +269,49 @@ def fn(a, b):
     return lines
 
 
-def only_sub_image_is_being_built(adr_map, app_output_dir):
-    if len(adr_map) == 2:
-        non_app_key = [non_app_key for non_app_key in adr_map.keys() if non_app_key != 'app'][0]
-        return app_output_dir == adr_map[non_app_key]['out_dir']
-    return False
-
-
-def write_pm_config(adr_map, app_output_dir):
+def write_pm_config(adr_map):
     pm_config_file = "pm_config.h"
     config_lines = get_config_lines(adr_map, "#define ", " ")
 
-    for area_name, area_props in adr_map.items():
+    for _, area_props in adr_map.items():
         area_props['pm_config'] = list.copy(config_lines)
         area_props['pm_config'].append("#define PM_ADDRESS 0x%x" % area_props['address'])
         area_props['pm_config'].append("#define PM_SIZE 0x%x" % area_props['size'])
         area_props['pm_config'].insert(0, get_header_guard_start(pm_config_file))
         area_props['pm_config'].append(get_header_guard_end(pm_config_file))
 
     # Store complete size/address configuration to all input paths
-    for area_name, area_props in adr_map.items():
+    for _, area_props in adr_map.items():
         if 'out_dir' in area_props.keys():
             write_pm_config_to_file(path.join(area_props['out_dir'], pm_config_file), area_props['pm_config'])
 
-    # Store to root app, but
-    if not only_sub_image_is_being_built(adr_map, app_output_dir):
-        write_pm_config_to_file(path.join(app_output_dir, pm_config_file), adr_map['app']['pm_config'])
-
 
 def write_pm_config_to_file(pm_config_file_path, pm_config):
     with open(pm_config_file_path, 'w') as out_file:
         out_file.write('\n'.join(pm_config))
 
 
-def write_kconfig_file(adr_map, app_output_dir):
+def write_kconfig_file(adr_map):
     pm_kconfig_file = "pm.config"
     config_lines = get_config_lines(adr_map, "", "=")
 
     # Store out dir and build dir
     for area_name, area_props in adr_map.items():
         if 'out_dir' in area_props.keys():
             config_lines.append('PM_%s_OUT_DIR="%s"' % (area_name.upper(), area_props['out_dir']))
+        if 'build_dir' in area_props.keys():
             config_lines.append('PM_%s_BUILD_DIR="%s"' % (area_name.upper(), area_props['build_dir']))
 
-    # Store output dir to app
-    config_lines.append('PM_APP_OUT_DIR "%s"' % app_output_dir)
-
     # Store complete size/address configuration to all input paths
-    for area_name, area_props in adr_map.items():
+    for _, area_props in adr_map.items():
         if 'out_dir' in area_props.keys():
             write_pm_config_to_file(path.join(area_props['out_dir'], pm_kconfig_file), config_lines)
 
-    # Store to root app
-    if not only_sub_image_is_being_built(adr_map, app_output_dir):
-        write_pm_config_to_file(path.join(app_output_dir, pm_kconfig_file), config_lines)
 
+def write_yaml_out_file(pm_config, path):
```

Comment:
`path` is defined by python, use other keyword.

Reply:
will change